integrations/acquisition/covidcast/test_is_latest_issue.py

```diff
@@ -0,0 +1,346 @@
+import pytest
+from prettytable import PrettyTable, from_db_cursor
+    
+# first party
+from delphi.epidata.acquisition.covidcast.logger import get_structured_logger
+from delphi_utils import Nans
+from delphi.epidata.client.delphi_epidata import Epidata
+from delphi.epidata.acquisition.covidcast.database import Database, CovidcastRow
+from delphi.epidata.acquisition.covidcast.covidcast_meta_cache_updater import main as update_covidcast_meta_cache
+import delphi.operations.secrets as secrets
+
+# py3tester coverage target (equivalent to `import *`)
+#  
+__test_target__ = 'delphi.epidata.acquisition.covidcast.database'
+
+
+# use the local instance of the Epidata API
+BASE_URL = 'http://delphi_web_epidata/epidata/api.php'
+
+nmv = Nans.NOT_MISSING.value
+class CovidcastLatestIssueTests(unittest.TestCase):
+  """Tests covidcast is_latest_issue caching."""
+
+  maxDiff = None #access full output of differences for debugging
+
+  def setUp(self):
+    """Perform per-test setup."""
+
+    # connect to the `epidata` database
+    cnx = mysql.connector.connect(
+        user='user',
+        password='pass',
+        host='delphi_database_epidata',
+        database='covid')
+    cur = cnx.cursor()
+
+    # clear all tables
+    cur.execute("truncate table signal_load")
+    cur.execute("truncate table signal_history")
+    cur.execute("truncate table signal_latest")
+    cur.execute("truncate table geo_dim")
+    cur.execute("truncate table signal_dim")
+    # reset the `covidcast_meta_cache` table (it should always have one row)
+    cur.execute('update covidcast_meta_cache set timestamp = 0, epidata = "[]"')
+    
+    cnx.commit()
+    cur.close()
+
+    # make connection and cursor available to the Database object
+    self._db = Database()
+    self._db._connection = cnx
+    self._db._cursor = cnx.cursor()
+
+    # use the local instance of the Epidata API
+    Epidata.BASE_URL = 'http://delphi_web_epidata/epidata/api.php'
+
+    # use the local instance of the epidata database
+    secrets.db.host = 'delphi_database_epidata'
+    secrets.db.epi = ('user', 'pass')
+
+  def tearDown(self):
+    """Perform per-test teardown."""
+    self._db._cursor.close()
+    self._db._connection.close()
+
+
+  def test_signal_latest(self):
+    #  def __init__(source, signal, time_type, geo_type, time_value, geo_value, 
+    #               value, stderr, sample_size, missing_value, missing_stderr, missing_sample_size, 
+    #               issue, lag, is_wip):
+    rows = [
+      CovidcastRow('src', 'sig', 'day', 'state', 20200414, 'pa', 
+                   1.5, 2.5, 3.5, nmv, nmv, nmv, 20200414, 0),
+      CovidcastRow('src', 'sig', 'day', 'state', 20200415, 'wa', 
+                   3, 2, 1, nmv, nmv, nmv, 20200415, 0)
+    ]
+    self._db.insert_or_update_bulk(rows)
+    self._db.run_dbjobs()
+    #preview
+    self._db._cursor.execute('''SELECT * FROM `signal_history`''')
+    self.totalRows = len(list(self._db._cursor.fetchall()))
```

Comment:
`totalRows` doesnt need to be an object member (test methods probbly shouldnt alter the TestCase object without a good reason)